eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -3238,6 +3238,126 @@ public function testMoveSubtreeUpdatesPathIdentificationString(): void
+
+        $sourceFolderContent = $this->publishContentWithParentLocation('SourceFolder', $mediaLocationId); // media/SourceFolder
+        $subFolderContent1 = $this->publishContentWithParentLocation('subFolderContent1', $sourceFolderContent->contentInfo->mainLocationId);
+        $subFolderContent2 = $this->publishContentWithParentLocation('subFolderContent2', $sourceFolderContent->contentInfo->mainLocationId);
+        $targetFolderContent = $this->publishContentWithParentLocation('targetFolder', $mediaLocationId); // media/SourceFolder
+
+        $contentService->hideContent($subFolderContent1->contentInfo);
+
+        // Move media/SourceFolder to media/TargetFolder/
+        $locationService->moveSubtree(
+            $sourceFolderContent->contentInfo->getMainLocation(),
+            $targetFolderContent->contentInfo->getMainLocation()
+        );
+
+        $movedLocation = $locationService->loadLocation($sourceFolderContent->contentInfo->mainLocationId);
+        $newParentLocation = $locationService->loadLocation($targetFolderContent->contentInfo->mainLocationId);
+
+        // Assert Moved Location remains visible ( only child is hidden )
+        $this->assertPropertiesCorrect(
+            [
+                'hidden' => false,
+                'invisible' => false,
+                'depth' => $newParentLocation->depth + 1,
+                'parentLocationId' => $newParentLocation->id,
+                'pathString' => $newParentLocation->pathString . $this->parseId('location', $movedLocation->id) . '/',
+            ],
+            $movedLocation
+        );
+        self::assertFalse($movedLocation->getContentInfo()->isHidden);
+
+        // Assert children of Moved location
+        foreach ([$subFolderContent1->contentInfo->getMainLocation(), $subFolderContent2->contentInfo->getMainLocation()] as $childLocation) {
+            $this->assertPropertiesCorrect(
+                [
+                    'hidden' => $childLocation === $subFolderContent1->contentInfo->getMainLocation(), // Only SubFolderContent1 should be hidden,
+                    'invisible' => $childLocation === $subFolderContent1->contentInfo->getMainLocation(), // Only SubFolderContent1 should be hidden
+                    'depth' => $movedLocation->depth + 1,
+                    'parentLocationId' => $movedLocation->id,
+                    'pathString' => $movedLocation->pathString . $this->parseId('location', $childLocation->id) . '/',
+                ],
+                $childLocation
+            );
+            self::assertEquals($childLocation === $subFolderContent1->contentInfo->getMainLocation(), $childLocation->getContentInfo()->isHidden);
+        }
+    }
+
+    /**
+     * Test that is_visible is set correct for children when moving a content which is hidden (location is not hidden).
+     *
+     * @covers \eZ\Publish\API\Repository\LocationService::moveSubtree
+     * depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testLoadLocation
```

Comment:
This one too

Reply:
Removed in a99fb158a9447ae40c7638048cd981a209de12d1